File: native/macos/SkillFlow/Sources/StarredRepos/StarredReposModels.swift
import Foundation

struct StarRepo: Codable, Identifiable, Equatable {
    let url: String
    let name: String
    let source: String
    let localDir: String
    let lastSync: Date
    let syncError: String?

    var id: String { url }

    var syncStatusText: String {
        if let error = syncError, !error.isEmpty { return error }
        let formatter = DateFormatter()
        formatter.dateFormat = "yyyy-MM-dd HH:mm"
        return formatter.string(from: lastSync)
    }
}

struct StarSkillEntry: Codable, Identifiable, Equatable {
    let name: String
    let path: String
    let subPath: String
    let repoUrl: String
    let repoName: String
    let source: String
    let logicalKey: String
    let installed: Bool
    let imported: Bool
    let updatable: Bool
    let pushed: Bool
    let pushedAgents: [String]

    var id: String { path }

    var statusText: String {
        if installed { return "Installed" }
        if imported { return "Imported" }
        return "New"
    }
}

struct StarredRepoURLParams: Encodable {
    let repoURL: String
}

struct StarredRepoCredentialsParams: Encodable {
    let repoURL: String
    let username: String
    let password: String
}

struct StarredImportParams: Encodable {
    let skillPaths: [String]
    let repoURL: String
    let category: String
}

struct StarredPushParams: Encodable {
    let skillPaths: [String]
    let agentNames: [String]
}

File: native/windows/SkillFlow/SkillFlow/StarredRepos/StarredReposModels.cs
using System.Text.Json.Serialization;

namespace SkillFlow.StarredRepos;

public sealed class StarRepo
{
    [JsonPropertyName("url")]
    public string Url { get; set; } = string.Empty;

    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("source")]
    public string Source { get; set; } = string.Empty;

    [JsonPropertyName("localDir")]
    public string LocalDir { get; set; } = string.Empty;

    [JsonPropertyName("lastSync")]
    public DateTimeOffset LastSync { get; set; }

    [JsonPropertyName("syncError")]
    public string SyncError { get; set; } = string.Empty;

    public string SyncStatusText => string.IsNullOrEmpty(SyncError)
        ? LastSync.ToString("yyyy-MM-dd HH:mm")
        : SyncError;
}

public sealed class StarSkillEntry
{
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("path")]
    public string Path { get; set; } = string.Empty;

    [JsonPropertyName("subPath")]
    public string SubPath { get; set; } = string.Empty;

    [JsonPropertyName("repoUrl")]
    public string RepoUrl { get; set; } = string.Empty;

    [JsonPropertyName("repoName")]
    public string RepoName { get; set; } = string.Empty;

    [JsonPropertyName("source")]
    public string Source { get; set; } = string.Empty;

    [JsonPropertyName("logicalKey")]
    public string LogicalKey { get; set; } = string.Empty;

    [JsonPropertyName("installed")]
    public bool Installed { get; set; }

    [JsonPropertyName("imported")]
    public bool Imported { get; set; }

    [JsonPropertyName("updatable")]
    public bool Updatable { get; set; }

    [JsonPropertyName("pushed")]
    public bool Pushed { get; set; }

    [JsonPropertyName("pushedAgents")]
    public List<string> PushedAgents { get; set; } = new();

    public string StatusText
    {
        get
        {
            if (Installed) return "Installed";
            if (Imported) return "Imported";
            return "New";
        }
    }
}

public sealed class StarredRepoURLParams
{
    [JsonPropertyName("repoURL")]
    public string RepoURL { get; set; } = string.Empty;
}

public sealed class StarredRepoCredentialsParams
{
    [JsonPropertyName("repoURL")]
    public string RepoURL { get; set; } = string.Empty;

    [JsonPropertyName("username")]
    public string Username { get; set; } = string.Empty;

    [JsonPropertyName("password")]
    public string Password { get; set; } = string.Empty;
}

public sealed class StarredImportParams
{
    [JsonPropertyName("skillPaths")]
    public List<string> SkillPaths { get; set; } = new();

    [JsonPropertyName("repoURL")]
    public string RepoURL { get; set; } = string.Empty;

    [JsonPropertyName("category")]
    public string Category { get; set; } = string.Empty;
}

public sealed class StarredPushParams
{
    [JsonPropertyName("skillPaths")]
    public List<string> SkillPaths { get; set; } = new();

    [JsonPropertyName("agentNames")]
    public List<string> AgentNames { get; set; } = new();
}
